### PR TITLE
Adds anyOf support

### PR DESCRIPTION
**Pull request by JbIPS**
To implement #49 issue

### CODE REVIEW

**Comment by mcollina** [comment]
CI is failing on node 4.

**Comment by JbIPS** [comment]
Damn, forgot rest params are not supported before 6.x. I'll find a workaround

**Comment by JbIPS** [comment]
All green \o/

File: index.js [hunk 1]
```diff
@@ -87,8 +88,19 @@ function build (schema, options) {
...
+    dependenciesName.push('fastSafeStringify')
+  }
+  if (hasAnyOf(schema)) {
+    dependencies.push(new Ajv())
+    dependenciesName.push('ajv')
+  }
+  if (dependencies.length > 0) {
+    dependenciesName.push(code)
+    return (Function.apply(null, dependenciesName).apply(null, dependencies))
   }
```
Comment (mcollina) [inline index.js]: I think you can remove the `if (dependencies.length > 0)`  check.

**Comment by JbIPS** [comment]
With Node 8.4.0
Fork
```
JSON.stringify array x 3,736 ops/sec ±0.73% (93 runs sampled)
fast-json-stringify array x 4,817 ops/sec ±0.69% (94 runs sampled)
fast-json-stringify-uglified array x 4,913 ops/sec ±0.76% (94 runs sampled)
JSON.stringify long string x 10,962 ops/sec ±0.49% (92 runs sampled)
fast-json-stringify long string x 10,968 ops/sec ±0.67% (95 runs sampled)
fast-json-stringify-uglified long string x 10,851 ops/sec ±1.00% (91 runs sampled)
JSON.stringify short string x 4,380,260 ops/sec ±0.71% (92 runs sampled)
fast-json-stringify short string x 9,730,481 ops/sec ±0.29% (93 runs sampled)
fast-json-stringify-uglified short string x 9,664,561 ops/sec ±0.87% (93 runs sampled)
JSON.stringify obj x 1,831,760 ops/sec ±0.38% (94 runs sampled)
fast-json-stringify obj x 4,448,195 ops/sec ±0.90% (94 runs sampled)
fast-json-stringify-uglified obj x 4,557,107 ops/sec ±0.24% (92 runs sampled)
```
Origin
```
JSON.stringify array x 3,827 ops/sec ±0.49% (93 runs sampled)
fast-json-stringify array x 4,660 ops/sec ±0.48% (91 runs sampled)
fast-json-stringify-uglified array x 4,733 ops/sec ±0.82% (92 runs sampled)
JSON.stringify long string x 10,939 ops/sec ±0.25% (93 runs sampled)
fast-json-stringify long string x 10,922 ops/sec ±0.61% (95 runs sampled)
fast-json-stringify-uglified long string x 10,972 ops/sec ±0.42% (92 runs sampled)
JSON.stringify short string x 4,306,996 ops/sec ±0.79% (95 runs sampled)
fast-json-stringify short string x 9,568,480 ops/sec ±0.71% (92 runs sampled)
fast-json-stringify-uglified short string x 9,758,963 ops/sec ±0.81% (92 runs sampled)
JSON.stringify obj x 1,804,430 ops/sec ±1.13% (95 runs sampled)
fast-json-stringify obj x 4,634,735 ops/sec ±0.79% (94 runs sampled)
fast-json-stringify-uglified obj x 4,712,696 ops/sec ±0.44% (95 runs sampled)
```

**Comment by mcollina** [comment]
Can you send another PR with the docs?

**Pull request merged**